core/topology/configuration_json.hxx

```diff
@@ -213,51 +213,76 @@ struct traits<couchbase::core::topology::configuration> {
                 } else if (name == "xdcrCheckpointing") {
-                    result.bucket_capabilities.insert(couchbase::core::bucket_capability::xdcr_checkpointing);
+                    result.capabilities.bucket.insert(couchbase::core::bucket_capability::xdcr_checkpointing);
                 } else if (name == "nodesExt") {
-                    result.bucket_capabilities.insert(couchbase::core::bucket_capability::nodes_ext);
+                    result.capabilities.bucket.insert(couchbase::core::bucket_capability::nodes_ext);
                 } else if (name == "xattr") {
-                    result.bucket_capabilities.insert(couchbase::core::bucket_capability::xattr);
+                    result.capabilities.bucket.insert(couchbase::core::bucket_capability::xattr);
                 } else if (name == "rangeScan") {
-                    result.bucket_capabilities.insert(couchbase::core::bucket_capability::range_scan);
+                    result.capabilities.bucket.insert(couchbase::core::bucket_capability::range_scan);
                 } else if (name == "subdoc.ReplicaRead") {
-                    result.bucket_capabilities.insert(couchbase::core::bucket_capability::replica_read);
+                    result.capabilities.bucket.insert(couchbase::core::bucket_capability::replica_read);
                 } else if (name == "nonDedupedHistory") {
-                    result.bucket_capabilities.insert(couchbase::core::bucket_capability::non_deduped_history);
+                    result.capabilities.bucket.insert(couchbase::core::bucket_capability::non_deduped_history);
+                } else if (name == "subdoc.ReplaceBodyWithXattr") {
+                    result.capabilities.bucket.insert(couchbase::core::bucket_capability::subdoc_replace_body_with_xattr);
+                } else if (name == "subdoc.DocumentMacroSupport") {
+                    result.capabilities.bucket.insert(couchbase::core::bucket_capability::subdoc_document_macro_support);
+                } else if (name == "subdoc.ReviveDocument") {
+                    result.capabilities.bucket.insert(couchbase::core::bucket_capability::subdoc_revive_document);
+                } else if (name == "dcp.IgnorePurgedTombstones") {
+                    result.capabilities.bucket.insert(couchbase::core::bucket_capability::dcp_ignore_purged_tombstones);
+                } else if (name == "preserve_expiry") {
+                    result.capabilities.bucket.insert(couchbase::core::bucket_capability::preserve_expiry);
+                } else if (name == "querySystemCollection") {
+                    result.capabilities.bucket.insert(couchbase::core::bucket_capability::query_system_collection);
+                } else if (name == "mobileSystemCollection") {
+                    result.capabilities.bucket.insert(couchbase::core::bucket_capability::mobile_system_collection);
+                } else if (name == "subdoc.ReplicaRead") {
```

Comment:
Duplicated `subdoc.ReplicaRead` check on L239 (causing build failure on alpine)

Reply:
Thank you for notice. I will update it, I think better leave the newer version of the enum value `subdoc_replica_read` as it reflects the underlying capability better (otherwise this error would not happen)